### PR TITLE
integration-cli: fix TestDaemonICC tests for newer iptables versions

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/pull/45789

Debian Woodworm ships with a newer version of iptables, which caused two tests to fail:

    === FAIL: amd64.integration-cli TestDockerDaemonSuite/TestDaemonICCLinkExpose (1.18s)
    docker_cli_daemon_test.go:841: assertion failed: false (matched bool) != true (true bool): iptables output should have contained "DROP.*all.*ext-bridge6.*ext-bridge6", but was "Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)\n pkts bytes target prot opt in out source destination \n 0 0 DOCKER-USER 0 -- * * 0.0.0.0/0 0.0.0.0/0 \n 0 0 DOCKER-ISOLATION-STAGE-1 0 -- * * 0.0.0.0/0 0.0.0.0/0 \n 0 0 ACCEPT 0 -- * ext-bridge6 0.0.0.0/0 0.0.0.0/0 ctstate RELATED,ESTABLISHED\n 0 0 DOCKER 0 -- * ext-bridge6 0.0.0.0/0 0.0.0.0/0 \n 0 0 ACCEPT 0 -- ext-bridge6 !ext-bridge6 0.0.0.0/0 0.0.0.0/0 \n 0 0 DROP 0 -- ext-bridge6 ext-bridge6 0.0.0.0/0 0.0.0.0/0 \n"
    --- FAIL: TestDockerDaemonSuite/TestDaemonICCLinkExpose (1.18s)

    === FAIL: amd64.integration-cli TestDockerDaemonSuite/TestDaemonICCPing (1.19s)
    docker_cli_daemon_test.go:803: assertion failed: false (matched bool) != true (true bool): iptables output should have contained "DROP.*all.*ext-bridge5.*ext-bridge5", but was "Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)\n pkts bytes target prot opt in out source destination \n 0 0 DOCKER-USER 0 -- * * 0.0.0.0/0 0.0.0.0/0 \n 0 0 DOCKER-ISOLATION-STAGE-1 0 -- * * 0.0.0.0/0 0.0.0.0/0 \n 0 0 ACCEPT 0 -- * ext-bridge5 0.0.0.0/0 0.0.0.0/0 ctstate RELATED,ESTABLISHED\n 0 0 DOCKER 0 -- * ext-bridge5 0.0.0.0/0 0.0.0.0/0 \n 0 0 ACCEPT 0 -- ext-bridge5 !ext-bridge5 0.0.0.0/0 0.0.0.0/0 \n 0 0 DROP 0 -- ext-bridge5 ext-bridge5 0.0.0.0/0 0.0.0.0/0 \n"
    --- FAIL: TestDockerDaemonSuite/TestDaemonICCPing (1.19s)

Both the `TestDaemonICCPing`, and `TestDaemonICCLinkExpose` test were introduced in dd0666e64f17329355c77aae1a2ac0fe2fe43402. These tests called `iptables` with the `-n` (`--numeric`) option, which prevents it from doing a reverse-DNS lookup as an optimization.

However, the `-n` option did not have an effect to the `prot` column before commit [da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa] (iptables < v1.8.9 or v1.8.8). Newer versions, such as the iptables version shipping with Debian Woodworm do, so we need to update the expected output for this version.

This patch removes the `-n` option, to keep the test more portable, also when run non-containerized, and removes the use of regular expressions to check the result, as these regular expressions were quite permissive (using `.*` wild- card matching). Instead, we're getting the

With this change;

make DOCKER_GRAPHDRIVER=vfs TEST_FILTER=TestDaemonICC TEST_IGNORE_CGROUP_CHECK=1 test-integration ...
--- PASS: TestDockerDaemonSuite (139.11s)
--- PASS: TestDockerDaemonSuite/TestDaemonICCLinkExpose (54.62s) --- PASS: TestDockerDaemonSuite/TestDaemonICCPing (84.48s)

[da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa]: https://git.netfilter.org/iptables/commit/?id=da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by cpuguy83 — COMMENTED** [review]
This looks like a great change.
Just a minor nit, and a question.

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -825,20 +834,29 @@ func (s *DockerDaemonSuite) TestDaemonICCLinkExpose(c *testing.T) {
...
 	defer d.Restart(c)
 
-	result := icmd.RunCommand("iptables", "-nvL", "FORWARD")
+	result := icmd.RunCommand("sh", "-c", "iptables -vL FORWARD | grep DROP")
 	result.Assert(c, icmd.Success)
-	regex := fmt.Sprintf("DROP.*all.*%s.*%s", bridgeName, bridgeName)
-	matched, _ := regexp.MatchString(regex, result.Combined())
-	assert.Equal(c, matched, true, fmt.Sprintf("iptables output should have contained %q, but was %q", regex, result.Combined()))
+	out := strings.TrimSpace(result.Stdout())
+	assert.Assert(c, is.Len(strings.Split(out, "\n"), 1), "only expected a single DROP rule %+v", strings.Split(out, "\n"))
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply integration-cli/docker_cli_daemon_test.go:849]: Ah, yes, that probably works. I just notice now I kept my debugging line at the end as well, so will remove that 🙈

File: integration-cli/docker_cli_daemon_test.go [hunk 2]
```diff
@@ -825,20 +834,29 @@ func (s *DockerDaemonSuite) TestDaemonICCLinkExpose(c *testing.T) {
...
+
+	// Column headers are stripped because of grep-ing, but should be:
+	//
+	//    pkts bytes target     prot opt in          out          source    destination
+	//       0     0 DROP       all  --  ext-bridge6 ext-bridge6  anywhere  anywhere
+	cols := strings.Fields(out)
+
+	expected := []string{"0", "0", "DROP", "all", "--", bridgeName, bridgeName, "anywhere", "anywhere"}
```
[the remark this answers is not in the document]
Comment (thaJeztah) [reply integration-cli/docker_cli_daemon_test.go:857]: (went looking if there's some magic flag to either print the information more structured, or to filter output, but there doesn't appear to be anything like that)

File: integration-cli/docker_cli_daemon_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (thaJeztah) [reply integration-cli/docker_cli_daemon_test.go]: ^ updated

**Review by cpuguy83 — APPROVED** [review]

**Review by akerouanton — APPROVED** [review]
LGTM

**Pull request merged**